src/factory/creative.js

```diff
@@ -1,15 +1,32 @@
+import {Creative} from 'iab-vast-model'
+import setUniversalAdId from '../factory/universal-ad-id'
+import createCreativeExtension from '../factory/creative-extension'
 import createLinear from './linear'
 import createNonLinearAds from './non-linear-ads'
 import createCompanionAds from './companion-ads'
 
 export default ($creative, options) => {
+  const creative = new Creative()
+  creative.id = $creative.id
+  creative.sequence = $creative.sequence
+  // VAST3 XSD specifies AdID and that seems to be what parsers have agreed on
+  creative.adID = $creative.AdID || $creative.adID || $creative.adId
+  creative.apiFramework = $creative.apiFramework
+  if ($creative.universalAdId) {
+    setUniversalAdId(creative.universalAdId, $creative.universalAdId)
+  }
+  if ($creative.creativeExtensions && $creative.creativeExtensions.creativeExtension) {
+    creative.creativeExtensions
+      .push(...$creative.creativeExtensions.creativeExtension.map(createCreativeExtension))
+  }
   if ($creative.linear) {
-    return createLinear($creative, options)
+    creative.linear = createLinear($creative, options)
```

Comment:
Might still change depending on iab-vast-model PR.

I also seem to remember that one of the reasons why we had to change this was because it's not an `else if`, since they can coexist?

Reply:
Yes, a <Linear> or a <NonLinearAds>. And an optional <CompanionAds>.

Reply:
Let's try again...

Yes, it either a Linear or a NonLinearAds. And an optional CompanionAds.